personhood/user/user.go

```diff
@@ -0,0 +1,306 @@
+package user
+
+import (
+	"fmt"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/cothority/v3/darc/expression"
+	"go.dedis.ch/cothority/v3/personhood/contracts"
+	"go.dedis.ch/onet/v3/log"
+	"go.dedis.ch/protobuf"
+	"golang.org/x/xerrors"
+	"strings"
+)
+
+// The User structure represents a Dynacred user as defined in the document
+// here: https://www.c4dt.org/wp-content/uploads/2020/09/DynaCred-101.pdf
+// It is a minimal implementation used for the bcadmin command, but can be
+// extended to more functionality, if needed.
+type User struct {
+	cl         *byzcoin.Client
+	credStruct contracts.CredentialStruct
+	// Signer is nil by default. It can be set by the user, so that more
+	// actions are available.
+	Signer   darc.Signer
+	CredIID  byzcoin.InstanceID
+	CredDarc darc.Darc
+}
+
+// New returns an initialized user.
+// It looks the credential instance up on ByzCoin and returns an error if
+// something went wrong: user not existing, wrong type of contract.
+func New(cl *byzcoin.Client, credIID byzcoin.InstanceID) (u User, err error) {
+	resp, err := cl.GetProof(credIID[:])
+	if err != nil {
+		err = xerrors.Errorf("while getting proof for credential instance: %v",
+			err)
+		return
+	}
+	buf, cid, did, err := resp.Proof.Get(credIID[:])
+	if err != nil {
+		err = xerrors.Errorf("reading proof: %v", err)
+		return
+	}
+	if cid != contracts.ContractCredentialID {
+		err = xerrors.Errorf("credIID points to wrong contract: %s instead of"+
+			" %s", cid, contracts.ContractCredentialID)
+		return
+	}
+	if err = protobuf.Decode(buf, &u.credStruct); err != nil {
+		err = xerrors.Errorf("while decoding CredentialStruct: %v", err)
+		return
+	}
```

Comment:
As in the remained of the function, you can `return u, xerrors.Errorf(...)`, it aids readbility.

Reply:
thanks - not sure anymore why I changed my mind in the middle of it...